src/ring_theory/algebra.lean

```diff
@@ -258,6 +258,24 @@ set.mem_image_of_mem (algebra_map R S) x.2
 
 end semiring
 
+section ring
+variables [comm_ring R] [comm_ring S] [ring A] [algebra R A]
+
+lemma mul_sub_algebra_map_commutes (x : A) (r : R) :
```

Comment:
I think this is basically a case of `commute.neg_right` (in `algebra.ring.basic`).

Reply:
`commute.neg_right` is still pretty far away from what I actually need. What do you suggest concretely?